runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8058,8 +8058,57 @@ done:;
 	VMINLINE VM_BytecodeAction
 	defaultvalue(REGISTER_ARGS_LIST)
 	{
+retry:
+		j9object_t newObject = NULL;
 		VM_BytecodeAction rc = EXECUTE_BYTECODE;
-		// TODO: Implement bytecode
+		const U_16 index = *(U_16*)(_pc + 1);
+		J9ConstantPool * const ramConstantPool = J9_CP_FROM_METHOD(_literals);
+		J9RAMClassRef * const ramCPEntry = ((J9RAMClassRef*)ramConstantPool) + index;
+		J9Class * volatile resolvedClass = ramCPEntry->value;
+		bool doInitializeClass = FALSE;
+
+		if ((NULL != resolvedClass) && J9_IS_J9CLASS_VALUETYPE(resolvedClass)) {
+			if (!VM_VMHelpers::classRequiresInitialization(_currentThread, resolvedClass)) {
+				j9object_t instance = _objectAllocate.inlineAllocateObject(_currentThread, resolvedClass);
+				if (NULL == instance) {
+					updateVMStruct(REGISTER_ARGS);
+					instance = _vm->memoryManagerFunctions->J9AllocateObject(_currentThread, resolvedClass, J9_GC_ALLOCATE_OBJECT_INSTRUMENTABLE);
+					VMStructHasBeenUpdated(REGISTER_ARGS);
+					if (J9_UNEXPECTED(NULL == instance)) {
+						rc = THROW_HEAP_OOM;
+						goto done;
+					}
+				}
+				newObject = instance;
+				goto done;
+			}
+			doInitializeClass = TRUE;
+		}
+
+		buildGenericSpecialStackFrame(REGISTER_ARGS, 0);
+		updateVMStruct(REGISTER_ARGS);
+		if (doInitializeClass) {
```

Comment:
I'm not sure all this complexity is worth it - can the initialize be part of the resolution?

Reply:
Given the first check is "resolved && valuetype", how about rewriting this in stages - check resolution, check valuetype, check class init, instead of nesting them together for no real savings.

Reply:
This does mean it now makes sense to move the VT check out of resolution completely since you have to do it here anyway.